### ISSUE TITLE
Latest version (24.04.3.1.1) does not send WOPI Lock/Unlock request

### ISSUE THREAD

**Issue by HFO4** [CollaboraOnline/online#9185]
### Describe the Bug
Latest version (24.04.3.1.1) does not send WOPI Lock/Unlock request while user open/save/close a file, even if `SupportsLocks` is set as `true` from WOPI server in `GetFileInfo` response. It did carry a `X-WOPI-Lock` header in `PUT` request while saving file, but this did not help on protecting file been modified by other apps.

I tried a very old version (22.05.9.2), the lock works as expected.

### Steps to Reproduce
1. Setup WOPI server with `SupportsLocks=true, SupportsUpdate=true, UserCanWrite=true`;
2. Open a document, no LOCK request sent from client;
3. Save the document, no LOCK request sent from the client;
4. Close the document, no UNLOCK request sent from the client.

### Expected Behavior
Send Lock request right after user open the document, maintain the lock valid by refreshing lock request as long as the document is opened; Send Unlock request after all user close the document and all changes are saved.

### Actual Behavior
No lock related request is sent.

### Additional Context
Attach log snippet of one user opens a document.
[coolwsd.log](https://github.com/CollaboraOnline/online/files/15493159/coolwsd.log)

**Comment by micbar** [comment]
@timar 

Unfortunately we are observing the same behavior using infinite scale. Collabora 24.04 stopped to do LOCK operations via WOPI.

Our wopiserver logs are logging

```json
{
  "time": "2024-06-11T18:36:31.136",
  "host": "fb11f09d0822",
  "level": "WARNING",
  "process": "wopiserver",
  "module": "wopiutils",
  "msg": "Returning conflict",
  "lockop": "Putfile",
  "user": "<redacted>",
  "filename": "ed6c204d-fbd7-486a-9e02-c25ed3233805/New file.odt",
  "token": "OmN8ir5RqEX99-Aaq9dU",
  "sessionId": "None",
  "lock": "cool-lockca0f53e2",
  "oldlock": "NA",
  "retrievedlock": "None",
  "fileage": "NA",
  "reason": "Cannot overwrite unlocked file"
}
```

The UI shows the following error:
![1718131458320](https://github.com/CollaboraOnline/online/assets/4378513/5f11a50f-3bbf-4bb6-9e45-36c7afb5e80f)

Unfortunately this is blocking us from using Collabora 24.04.

It works fine with Collabora 23.04

**Comment by mmeeks** [comment]
Thanks for the report; I've filed a support ticket for you - @annanazarayan will chase that =)

**Comment by annanazarayan** [comment]
@micbar currently the ticket status is in progress, it happens in certain scenarios hopefully we will have it fixed soon.

**Comment by thebearon** [comment]
To note, the fix is part of CODE 24.0.4.5-1, and will come with the COOL 24.04.5 update due in a few weeks.
